src/tasks/FilterFeedTask.ts

```diff
@@ -0,0 +1,83 @@
+import { MentionRegistry } from '../mentions/MentionRegistry';
+import { FilterFeedConfig } from '../BotConfig';
+import { Client, TextChannel } from 'discord.js';
+import * as log4js from 'log4js';
+import Task from './Task';
+import JiraClient = require( 'jira-connector' );
+
+export default class FilterFeedTask extends Task {
+	public static logger = log4js.getLogger( 'FilterFeed' );
+
+	private client: Client;
+	private jira: JiraClient;
+	private channel: string;
+	private jql: string;
+	private title: string;
+
+	private knownTickets: Set<string>;
+
+	constructor( { channel, jql, title }: FilterFeedConfig, client: Client ) {
+		super();
+
+		this.channel = channel;
+		this.jql = jql;
+		this.title = title;
+		this.client = client;
+
+		this.jira = new JiraClient( {
+			host: 'bugs.mojang.com',
+			strictSSL: true,
+		} );
+	}
+
+	public async run(): Promise<void> {
+		let upcomingTickets: string[];
+
+		try {
+			const searchResults = await this.jira.search.search( {
+				jql: this.jql,
+				fields: ['key'],
+			} );
+
+			if ( !searchResults.issues ) {
+				FilterFeedTask.logger.error( 'Error: no issues returned by JIRA' );
+				return;
+			}
+
+			upcomingTickets = searchResults.issues.map( ( { key } ) => key );
+		}
+		catch ( err ) {
+			FilterFeedTask.logger.error( err );
+			return;
+		}
+
+		if ( this.knownTickets ) {
+			const unknownTickets = upcomingTickets.filter( key => !this.knownTickets.has( key ) );
+
+			if ( unknownTickets.length > 0 ) {
+				try {
+					const embed = await MentionRegistry.getMention( unknownTickets ).getEmbed();
+					embed
+						.setTitle(
+							this.title
+								.replace( /\{\{num\}\}/g, unknownTickets.length.toString() )
+						)
+						.setTimestamp( new Date() );
+
+					const channel = this.client.channels.get( this.channel );
+					if ( channel instanceof TextChannel ) {
+						channel.send( embed );
+					}
+					else {
+						throw `Expected ${this.channel} to be a TextChannel`;
+					}
+				}
+				catch ( err ) {
+					FilterFeedTask.logger.error( err );
+					return;
+				}
+			}
+		}
+		upcomingTickets.forEach( key => this.knownTickets.add( key ) );
```

Comment:
Is there any particular advantage of `upcomingTickets.forEach` over `for (obj of upcomingTickets)`?

Reply:
No, just my personal preference here. Will changed to `forof`.